src/gc.c

```diff
@@ -2461,6 +2458,10 @@ void jl_gc_init(void)
     last_long_collect_interval = default_collect_interval;
     allocd_bytes = -default_collect_interval;
 
+    char *max_pages_str = getenv("JULIA_GC_MAX_PAGES");
```

Comment:
Should this maybe be specified in terms of total memory rather than pages? Internally, it's used to figure out a number of pages but it seems like this should be in the same units that `ulimit -v` is specified.